### PR TITLE
Update model-contracts.md for supported models

### PR DESCRIPTION
**Pull request by nathangriffiths-cdx**
## What are you changing in this pull request and why?

Update to show that incremental models with on_schema_change: fail are actually supported (only on_schema_change: append_new_columns is mentioned but this was changed here https://github.com/dbt-labs/dbt-core/pull/8006)

Reformatted the list to make it clearer what is and isn't supported.

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 1]
```diff
@@ -28,10 +28,18 @@ While this is ideal for quick and iterative development, for some models, consta
 ## Where are contracts supported?
 
 At present, model contracts are supported for:
-- SQL models. Contracts are not yet supported for Python models.
-- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`). Views offer limited support for column names and data types, but not `constraints`. Contracts are not supported for `ephemeral`-materialized models.
+- SQL models. 
+- Models materialized as one of the following
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Models materialized as one of the following:
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 2]
```diff
@@ -28,10 +28,18 @@ While this is ideal for quick and iterative development, for some models, consta
...
+    - `table`
+    - `view`. Views offer limited support for column names and data types, but not `constraints`.
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
    - `view` &mdash; Views offer limited support for column names and data types, but not `constraints`.
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 3]
```diff
@@ -28,10 +28,18 @@ While this is ideal for quick and iterative development, for some models, consta
...
+    - `incremental` (with `on_schema_change: append_new_columns` or `on_schema_change: fail`).
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
    - `incremental` &mdash; with `on_schema_change: append_new_columns` or `on_schema_change: fail`.  
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 4]
```diff
@@ -28,10 +28,18 @@ While this is ideal for quick and iterative development, for some models, consta
...
+- SQL models. 
+- Models materialized as one of the following:
+    - `table`
+    - `view` &mdash; Views offer limited support for column names and data types, but not `constraints`.
+    - `incremental` &mdash; with `on_schema_change: append_new_columns` or `on_schema_change: fail`.  
 - Certain data platforms, but the supported and enforced `constraints` vary by platform.
 
+Model contracts are **not** supported for:
+- Python models.
+- `ephemeral`-materialized SQL models.
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md:40]:
```suggestion
- `ephemeral`&mdash; materialized SQL models.
```

**Comment by matthewshaver** [comment]
Thank you @nathangriffiths-cdx Made some minor alterations based on style guide but this is a nice improvement and we appreciate it.

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 5]
```diff
@@ -28,10 +28,18 @@ While this is ideal for quick and iterative development, for some models, consta
...
+- SQL models. 
+- Models materialized as one of the following:
+    - `table`
+    - `view` &mdash; Views offer limited support for column names and data types, but not `constraints`.
+    - `incremental` &mdash; with `on_schema_change: append_new_columns` or `on_schema_change: fail`.  
 - Certain data platforms, but the supported and enforced `constraints` vary by platform.
 
+Model contracts are **not** supported for:
+- Python models.
+- `ephemeral`&mdash; materialized SQL models.
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- `ephemeral`-materialized SQL models.
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 6]
```diff
@@ -28,10 +28,18 @@ While this is ideal for quick and iterative development, for some models, consta
...
-- SQL models. Contracts are not yet supported for Python models.
-- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`). Views offer limited support for column names and data types, but not `constraints`. Contracts are not supported for `ephemeral`-materialized models.
+- SQL models. 
+- Models materialized as one of the following:
+    - `table`
+    - `view` &mdash; Views offer limited support for column names and data types, but not `constraints`.
+    - `incremental` &mdash; with `on_schema_change: append_new_columns` or `on_schema_change: fail`.  
 - Certain data platforms, but the supported and enforced `constraints` vary by platform.
 
+Model contracts are **not** supported for:
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
Model contracts are _not_ supported for:
```

**Review by dloubser-wownz — APPROVED** [review]

**Review by matthewshaver — APPROVED** [review]

**Pull request merged**